### ISSUE TITLE
S3626: Possible false positive on return statement 

### ISSUE THREAD

**Issue by JelkeH** [SonarSource/sonar-dotnet#7098]
### Description
The repro code below produces warning [S3626](https://sonarsource.github.io/rspec/#/rspec/S3626/csharp). 
Although  it is technically correct, I doubt if it should be issued here.
It adds clarity and protect against unintended statements

### Repro steps
```cs
public class C
{
    public void M(List<int> foo)
    {
        Bar(foo);
        Baz(foo);
        return; // FP S3626: Remove this redundant jump.
        
        static void Bar(List<int> foo)
        {
            ...
        }
        
        static void Baz(List<int> foo)
        {
            ...
        }
    }
}
```

The return above adds clarity to the reader that there are no statements expected in `M` between the functions and protect against something like:

```cs
public class C
{
    public void M(List<int> foo)
    {
        Bar(foo);
        Baz(foo);
        return; // FP S3626: Remove this redundant jump.
        
        static void Bar(List<int> foo)
        {
            ...
        }
        
        foo.Clear(); //<----
        
        static void Baz(List<int> foo)
        {
            ...
        }
    }
}
```

### Expected behavior
[S3626](https://sonarsource.github.io/rspec/#/rspec/S3626/csharp) is not reported in the code above.

### Actual behavior
[S3626](https://sonarsource.github.io/rspec/#/rspec/S3626/csharp) is arguably incorrectly reported.

### Known workarounds
Suppress [S3626](https://sonarsource.github.io/rspec/#/rspec/S3626/csharp).

**Comment by pavel-mikula-sonarsource** [comment]
Hi @JelkeH,

As you mentioned, it's technically correct. So it behaves as intended.

What you're trying to prevent sounds like a new rule idea - something like "do not put statements between local functions". I've created #7101 to capture the idea.

**Comment by rcdailey** [comment]
@pavel-mikula-sonarsource 

> As you mentioned, it's technically correct. So it behaves as intended.

I'd like to add my thoughts here. While technically correct, this false-positive is caused by the analyzer's ignorance of the programmer's intent. The `return` is not redundant. It serves a functional purpose, and that is to separate the local methods from behavioral logic in the parent method.

I think that this issue still has merit. I could see a simple change made on the Sonar side that checks if local methods are defined after a `return` statement. And if so, it would not raise S3626 as an issue.

**Comment by lg2de** [comment]
I want to support this request to add an exception for a `return` before local functions.
One issue I face often in this context is the result of other analyzers, e.g. JetBrains Rider: https://www.jetbrains.com/help/rider/SeparateLocalFunctionsWithJumpStatement.html
